src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java

```diff
@@ -138,14 +138,26 @@ public void verify(String token, String nonce, Integer maxAuthenticationAge) thr
 
         // Org verification
         if (this.organization != null) {
-            String orgClaim = decoded.getClaim("org_id").asString();
-            if (isEmpty(orgClaim)) {
-                throw new IdTokenValidationException("Organization Id (org_id) claim must be a string present in the ID token");
+            String org = this.organization.trim();
+            if (org.startsWith("org_")) {
+                // org ID
+                String orgClaim = decoded.getClaim("org_id").asString();
+                if (isEmpty(orgClaim)) {
+                    throw new IdTokenValidationException("Organization Id (org_id) claim must be a string present in the ID token");
+                }
+                if (!this.organization.equals(orgClaim)) {
+                    throw new IdTokenValidationException(String.format("Organization (org_id) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", this.organization, orgClaim));
+                }
+            } else {
+                // org name
+                String orgNameClaim = decoded.getClaim("org_name").asString();
+                if (isEmpty(orgNameClaim)) {
+                    throw new IdTokenValidationException("Organization name (org_name) claim must be a string present in the ID token");
+                }
+                if (!org.equalsIgnoreCase(orgNameClaim)) {
```

Comment:
I've made a mistake here @jimmyjames, please see the internal discussion. This should not strictly be an ignore-case check but should instead be a check against the claim value, and lowercased `org` (subtle difference).